Move some settings into environment variables (#229)
sapcc/limes: docs/operators/config.md

| `LIMES_DB_CONNECTION_OPTIONS` | *(optional)* | Database connection options. |
| `OS_...` | *(required)* | A full set of OpenStack auth environment variables for Limes's service user. See [documentation for openstackclient](https://docs.openstack.org/python-openstackclient/latest/cli/man/openstack.html) for details. |

## Environment variables for `limes serve` only

| Variable | Default | Description |
| --- | --- | --- |
| `LIMES_API_LISTEN_ADDRESS` | `:80` | Bind address for the HTTP API exposed by this service, e.g. `127.0.0.1:80` to bind only on one IP, or `:80` to bind on all interfaces and addresses. |
| `LIMES_API_POLICY_PATH` | `/etc/limes/policy.yaml` | Path to the oslo.policy file that describes authorization behavior for this service. Please refer to the [OpenStack documentation on policies][policy] for syntax reference. This repository includes an [example policy][ex-pol] that can be used for development setups, or as a basis for writing your own policy. For `:raise`, `:raise_lowpriv`, `:lower` and `:set_rate_limit` policies, the object attribute `%(service_type)s` is available to restrict editing to certain service types. |

> | `LIMES_OPA_DOMAIN_QUOTA_POLICY_PATH` | yes, when `opa` section is given | Like `LIMES_OPA_PROJECT_QUOTA_POLICY_PATH`, but this policy is used for validating domain quota changes instead. |
> | `LIMES_OPA_PROJECT_QUOTA_POLICY_PATH` | yes, when `opa` section is given | Path to a Rego file. This policy will be evaluated whenever a user tries to change a project quota. If the policy reports any violations, the PUT will fail and the generated error messages will be reported to the user. See [../example-quota-policy.rego](../example-quota-policy.rego) for an example illustrating the expected structure of the policy. |

Author: ```suggestion
| `LIMES_OPA_DOMAIN_QUOTA_POLICY_PATH` | yes, when `opa` section is given | Path to a Rego file. This policy will be evaluated whenever a user tries to change a domain quota. If the policy reports any violations, the PUT will fail and the generated error messages will be reported to the user. See [../example-quota-policy.rego](../example-quota-policy.rego) for an example illustrating the expected structure of the policy. |
| `LIMES_OPA_PROJECT_QUOTA_POLICY_PATH` | yes, when `opa` section is given | Like `LIMES_OPA_PROJECT_QUOTA_POLICY_PATH`, but this policy is used for validating project quota changes instead. |
```

Reviewer: Done.

| `LIMES_COLLECTOR_DATA_METRICS_SKIP_ZERO` | `false` | If set to `true`, data metrics will only be emitted for non-zero values. In large deployments, this can substantially reduce the amount of timeseries emitted. |

## Configuration

Authentication to keystone is done via the standard `OS_*` environment variables.
Also some key settings and paths are set via environment variables to allow easy local testing.

| Variable | Required | Description |
| --- | --- | --- |
| `OS_AUTH_URL` | yes | URL for Keystone v3 API in this cluster. Should end in `/v3`. Other Keystone API versions are not supported. |
| `OS_USERNAME` | yes | Limes service user. |
| `OS_USER_DOMAIN_NAME` | yes | Domain containing Limes service user. |
| `OS_PROJECT_NAME` | yes | Project where Limes service user has access. |
| `OS_PROJECT_DOMAIN_NAME` | yes | Domain containing that project. |
| `OS_PASSWORD` | yes | Password for Limes service user. |
| `OS_REGION_NAME` | no | In multi-region OpenStack clusters, this selects the region to work on. |

> | `OS_INTERFACE` | no | The endpoint type Limes should use for the OpenStack services. |

Reviewer: There are lots of other variations of configuration that are supported by gophercloud/utils/clientconfig. For other services, I usually point to the upstream documentation for openstackclient, see e.g. https://github.com/sapcc/tenso/blob/183c721f334e86cab344afb5c3018acc1c28226d/README.md?plain=1#L63

| `LIMES_AUTHORITATIVE` | no | If set to `true`, the collector will write the quota from its own database into the backend service whenever scraping encounters a backend quota that differs from the expectation. This flag is strongly recommended in production systems to avoid divergence of Limes quotas from backend quotas, but should be used with care during development. |
| `LIMES_CONSTRAINTS_PATH` | no | Path to a YAML file containing the quota constraints for this cluster. See [*quota constraints*](constraints.md) for details. |

> | `LIMES_OPA_DOMAIN_QUOTA_POLICY_PATH` | yes, when `opa` section is given | Like `opa.project_quota_policy_path`, but this policy is used for validating domain quota changes instead. |

Reviewer: ```suggestion
| `LIMES_OPA_DOMAIN_QUOTA_POLICY_PATH` | yes, when `opa` section is given | Like `LIMES_OPA_PROJECT_QUOTA_POLICY_PATH`, but this policy is used for validating domain quota changes instead. |
```

| `LIMES_OPA_PROJECT_QUOTA_POLICY_PATH` | yes, when `opa` section is given | Path to a Rego file. This policy will be evaluated whenever a user tries to change a project quota. If the policy reports any violations, the PUT will fail and the generated error messages will be reported to the user. See [../example-quota-policy.rego](../example-quota-policy.rego) for an example illustrating the expected structure of the policy. |

> The configuration file options relating to the targeted OpenStack cluster. A minimal config file could look like this:

Reviewer: While we're at it, we should fix the broken grammar in this line.
```suggestion
Furthermore, a YAML file must be provided that describes things like the set of available backend services and the quota/capacity scraping behavior. A minimal config file could look like this:
```
